src/rgw/rgw_lua_background.h

```diff
@@ -0,0 +1,68 @@
+  #pragma once
+  #include "common/dout.h"
+  #include "rgw_common.h"
+  #include <string>
+  #include "rgw_lua_utils.h"
+
+  namespace rgw::lua {
+
+  //Interval between each execution of the script
+  //is set to 5 seconds, this setting is configurable
+  //using RGW.setoff from within the lua script
+  constexpr const int INIT_SET_OFF = 5;
+ 
+  //Writeable meta table named RGW
+  using BackgroundMap = std::unordered_map<std::string, std::string>;
+  struct RGWTable : StringMapMetaTable<BackgroundMap,
```

Comment:
please use your own map type, that has a mutex around the relevant functions
for simplicity, you can wrap it with your class, and expose only the methods used here (when you add the mutex protection)